File: server/app/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using app.Enums;

namespace app.Models;

public class Product{
	[Key] public required string Barcode{ get; init; }

	public required string CatalogNumber{ get; init; }
	
	public required string Name{ get; set; }

	public required string Artist{ get; set; }

	public required string ImageUrl{ get; set; }

	public required decimal? Price{ get; set; }

	public required ProductType Type{ get; init; }
	
	public ICollection<Track> Tracklist{ get; set; } = [];

	[StringLength(11)] public required string Runtime{ get; init; }

	[StringLength(10)] public required string ReleaseDate{ get; init; }
	
	public required bool InStock{ get; set; } 
}

File: client/src/models/Product.ts
import Track from "./Track.ts";

export default class Product {
    name: string;
    artist: string;
    imageUrl: string;
    releaseDate: string;
    type: number;
    barcode: string;
    catalogNumber: string;
    runtime: string;
    price: string | null;
    tracklist: Array<Track>;
    inStock: boolean;

    constructor(jsonData: any) {
        ({
            barcode: this.barcode,
            catalogNumber: this.catalogNumber,
            name: this.name,
            artist: this.artist,
            imageUrl: this.imageUrl,
            price: this.price,
            runtime: this.runtime,
            type: this.type,
            tracklist: this.tracklist,
            releaseDate: this.releaseDate,
            inStock: this.inStock
        } = jsonData);
    }
    
    public static evaluateType = (type: number) => {
        switch (type) {
            case 0:
                return "Vinyl"
            case 1:
                return "Cassette"
            case 2:
                return "CD"
            default:
                return "Unknown"
        }
    }

} 